agent/util-scripts/pbench-tool-meister

```diff
@@ -0,0 +1,1019 @@
+    handling argument processing, logging setup, initial connection to
+    Redis(), fetch and validation of operational paramters from Redis(), and
+    then the daemonization of the ToolMeister operation.
+
+    Arguments:  argv - a list of parameters
+
+    Returns 0 on success, > 0 when an error occurs.
+
+    """
+    PROG = os.path.basename(argv[0])
+    pbench_bin = os.path.dirname(os.path.dirname(argv[0]))
+
+    try:
+        redis_host = argv[1]
+        redis_port = argv[2]
+        param_key = argv[3]
+    except IndexError as e:
+        print(f"Invalid arguments: {e}", file=sys.stderr)
+        return 1
+
+    global tar_path
+    tar_path = find_executable("tar")
+    if tar_path is None:
+        print(f"External 'tar' executable not found.", file=sys.stderr)
+        return 2
+
+    logger = logging.getLogger(PROG)
+    fh = logging.FileHandler(f"{param_key}.log")
+    if os.environ.get("_PBENCH_UNIT_TESTS"):
+        fmtstr = "%(levelname)s %(name)s %(funcName)s -- %(message)s"
+    else:
+        fmtstr = (
+            "%(asctime)s %(levelname)s %(process)s %(thread)s"
+            " %(name)s %(funcName)s %(lineno)d -- %(message)s"
+        )
+    fhf = logging.Formatter(fmtstr)
+    fh.setFormatter(fhf)
+    if os.environ.get("_PBENCH_TOOL_MEISTER_LOG_LEVEL") == "debug":
+        log_level = logging.DEBUG
+    else:
+        log_level = logging.INFO
+    fh.setLevel(log_level)
+    logger.addHandler(fh)
+    logger.setLevel(log_level)
+
+    try:
+        redis_server = redis.Redis(host=redis_host, port=redis_port, db=0)
+    except Exception as e:
+        logger.error(
+            "Unable to construct Redis client, %s:%s: %s", redis_host, redis_port, e
+        )
+        return 3
+
+    try:
+        params_raw = redis_server.get(param_key)
+        if params_raw is None:
+            logger.error('Parameter key, "%s" does not exist.', param_key)
+            return 4
+        logger.info("params_key (%s): %r", param_key, params_raw)
+        params_str = params_raw.decode("utf-8")
+        params = json.loads(params_str)
+        # Validate the tool meister parameters without constructing an object
+        # just yet, as we want to make sure we can talk to the redis server
+        # before we go through the trouble of daemonizing below.
+        ToolMeister.fetch_params(params)
+    except Exception as exc:
+        logger.error(
+            "Unable to fetch and decode parameter key, '%s': %s", param_key, exc
+        )
+        return 5
+    else:
+        redis_server.connection_pool.disconnect()
+        del redis_server
+
+    # Before we daemonize, flush any data written to stdout or stderr.
+    sys.stderr.flush()
+    sys.stdout.flush()
+
+    ret_val = 0
+    pidfile_name = f"{param_key}.pid"
+    pfctx = pidfile.PIDFile(pidfile_name)
+    with open(f"{param_key}.out", "w") as sofp, open(
+        f"{param_key}.err", "w"
+    ) as sefp, daemon.DaemonContext(
```

Comment:
Nice. The wrapping of the second open() is a bit weird; and I'm not sure the `with` serves much purpose as the daemon will terminate at the end of the try anyway, but it looks like covered bases.

Reply:
The wrapping is `black`'s take on proper formatting. :(